ledger/simulation/simulator.go

```diff
@@ -36,6 +36,11 @@ type simulatorLedger struct {
 	start basics.Round
 }
 
+// SimulatorConfig packs simulation related configurations, that overlaps the configurations in real transactions.
+type SimulatorConfig struct {
```

Comment:
Two minor things:
* Every other package will refer to this with the `simulation.` prefix, so I would suggest dropping the `Simulator` type prefix here
* I think it would be nice to establish consistency with the REST API's `SimulationRequest` -- so maybe make this a `Request` that contains `LiftLogLimits` and also the signed txn array? That way the only argument that the `Simulator.Simualte` function needs is now a `Request`

Reply:
right, the symmetric structure should still hold, and easier to keep track of things around.

Reply:
see [refactor Simulate interface, remove unnecessary codec struct, remove …](https://github.com/algorand/go-algorand/pull/5247/commits/61d8b7e19f08848a812d01c8933950740c97b60d)